### PR TITLE
Clicking in center of image goes to next

### PR DESCRIPTION
**Pull request by marekdedic**
This makes it so that clicking in the left third of an image moves you to the previous one, and the right two thirds moves you to the next one (as opposed to half and half)

This to me makes more sense, before if you click in the middle of the image, you expect to go to the next one...

### CODE REVIEW

File: src/imagelightbox.js [hunk 1]
```diff
@@ -517,7 +517,7 @@
                             return true;
                         }
                         var posX = (e.pageX || e.originalEvent.pageX) - e.target.offsetLeft;
-                        if (e.target.width / 2 > posX) {
+                        if (e.target.width / 3 > posX) {
```
Comment (rejas) [inline src/imagelightbox.js]: could you make it configurable? so people could even make a click on the whole image go to the next one? something like a clickPercentageForNextImage parameter (set to 50% maybe). But find a bette rname please :-)

Comment (marekdedic) [reply src/imagelightbox.js]: Hmm, I don't know... It seems to me to be a bit unnecessary to offer the user the whole percentage. But I agree that someone might want the whole image to go next - maybe just these two options?

Comment (rejas) [reply src/imagelightbox.js]: I'd still favor one option instead of two. This way people can decide for themselves :-)

Comment (marekdedic) [reply src/imagelightbox.js]: Umm, I'm not sure we understand each other - I was proposing to let user choose just between "whole image goes to next" and "Left side (1/3) to previous, right side(2/3) to next".

**Comment by marekdedic** [comment]
Hey, I tried to conceptualize what I meant - add the option to make the whole image go to next or just the right two thirds.

If you have a better name for the option, that would be great.

**Comment by marekdedic** [comment]
Hey @rejas any update please? I would like to get this "out of the door" and focus on other things ;)

**Comment by rejas** [comment]
Sorry for being silent in you. Will take a look at it tomorrow after out holidays here.
Am 30.05.19, 16:06, "Marek Dědič" <notifications@github.com> schrieb:
Hey @rejas any update please? I would like to get this "out of the door" and focus on other things ;)

—You are receiving this because you were mentioned.Reply to this email directly, view it on GitHub, or mute the thread.
[
{
"@context": "http://schema.org",
"@type": "EmailMessage",
"potentialAction": {
"@type": "ViewAction",
"target": "https://github.com/rejas/imagelightbox/pull/196?email_source=notifications\u0026email_token=AAKAWC5GPCUSA27ROPUIFLTPX7NNFA5CNFSM4HDJD32KYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODWSNKEY#issuecomment-497341715",
"url": "https://github.com/rejas/imagelightbox/pull/196?email_source=notifications\u0026email_token=AAKAWC5GPCUSA27ROPUIFLTPX7NNFA5CNFSM4HDJD32KYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODWSNKEY#issuecomment-497341715",
"name": "View Pull Request"
},
"description": "View this Pull Request on GitHub",
"publisher": {
"@type": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Comment by paxperscientiam** [comment]
> This makes it so that clicking in the left third of an image moves you to the previous one, and the right two thirds moves you to the next one (as opposed to half and half)
> 
> This to me makes more sense, before if you click in the middle of the image, you expect to go to the next one...

This, I think, is where Internationalization can come into play. Your suggestion makes sense to you and I as Westerners, but for people who read right-to-left the opposite would be the case.

So, before touching that logic, I would take a step back and consider what Internationalization means in terms of this project. The only relevant factor, I think, would be "read direction."

For example, for LTR, we'd use the right 2/3 area to go right and the opposite for RTL.

When I transcribe [codrops/BookBlock](https://github.com/codrops/BookBlock) into Typescript, I didn't touch the read direction stuff; however, you might find it useful as a reference.
https://github.com/paxperscientiam/BookBlock/tree/modern

**Comment by marekdedic** [comment]
Ha, but then we would need to do the animation the other way around as well, right?

**Comment by paxperscientiam** [comment]
Off the top of my head, I don't recall if the animation is completely handled by JS or CSS. If the former, then solved programmatically. If the latter, then solved with a separate class applied programmatically.

**Comment by marekdedic** [comment]
I am merging the straightforward part of this PR, leaving the rest for discussion in #369 and #370

**Pull request merged**